TASK: Fix requirements of dev packages

dlubitz opened the pull request:
The following packages have only a automatic deployment on the first release of each minor version (x.y.0):
* neos/welcome
* neos/behat
* neos/buildessentials

After the first release in the minor version, we need to do tag new versions manually and the version numbers are not in sync with the flow release versions. 

To ensure, a matching version of the packages listed above, we need to set the constraints to `~x.y.0`, which allows to install at least the automatically released version, but also newer versions that are tagged manually.

See also: https://neos-project.slack.com/archives/C04PYL8H3/p1720790338419519

dlubitz committed:
TASK: Fix requirements of dev packages

dlubitz committed:
TASK: Fix requirements of dev packages

kitsunet commented:
ah shoot, I was about to ✅this but then I realized... what about 9.0... because on beta11 it would require ~9.0.0 which does not exist I think? Or did we tag those already with a 9.0.0 otherwise we would need IDK a special case for beta versions or so?

dlubitz commented:
> ah shoot, I was about to ✅this but then I realized... what about 9.0... because on beta11 it would require ~9.0.0 which does not exist I think? Or did we tag those already with a 9.0.0 otherwise we would need IDK a special case for beta versions or so?

😭

kitsunet committed:
Use dev versions if not releasing stable version

Include dev dependencies on non collection
packages when a stability flag (alpha/beta/rc) was
set for the release version as we cannot be sure
we have (or want) a tagged version for those.

kitsunet commented:
this should be better :)
